src/Microsoft.VisualStudio.Composition/RuntimeComposition.cs

```diff
@@ -21,12 +21,22 @@ public class RuntimeComposition : IEquatable<RuntimeComposition>
         private readonly IReadOnlyDictionary<string, IReadOnlyCollection<RuntimeExport>> exportsByContractName;
         private readonly IReadOnlyDictionary<TypeRef, RuntimeExport> metadataViewsAndProviders;
 
+        public const string InvalidCompositionException = nameof(InvalidCompositionException);
+
         private RuntimeComposition(IEnumerable<RuntimePart> parts, IReadOnlyDictionary<TypeRef, RuntimeExport> metadataViewsAndProviders, Resolver resolver)
         {
             Requires.NotNull(parts, nameof(parts));
             Requires.NotNull(metadataViewsAndProviders, nameof(metadataViewsAndProviders));
             Requires.NotNull(resolver, nameof(resolver));
 
+            if (!parts.Any() || !metadataViewsAndProviders.Any())
```

Comment:
Do we already have a unit test to verify that an empty `ComposableCatalog` can still compose and turn into an `ExportProvider` that doesn't throw here? If not, please add one.